pysat/utils/_core.py

```diff
@@ -197,7 +198,24 @@ def load_netcdf4(fnames=None, strict_meta=False, file_format='NETCDF4',
         'NETCDF3_CLASSIC', 'NETCDF3_64BIT', 'NETCDF4_CLASSIC', or 'NETCDF4'.
         (default='NETCDF4')
     epoch_name : str
-        Data key for time variable (default='Epoch')
+        Data key for epoch variable.  The epoch variable is expected to be an
+        array of interger or float values denoting time elapsed from an origin
+        specified by `epoch_origin` with units specified by `epoch_unit`. This
+        epoch variable will be converted to a `DatetimeIndex` for consistency
+        across pysat instruments.  (default='Epoch')
+    epoch_unit : str
+        The unit of the epoch variable ('D', 's', 'ms', 'us', 'ns').
+        (default='ms')
+    epoch_origin : str or timestamp-convertable
+        Origin of epoch calculation, following convention for
+        `pandas.to_datetime`.  Accepts timestamp-convertable objects, as well as
+        two specific strings for commonly used calendars.  These conversions are
+        handled by `pandas.to_datetime`.
+        If ‘unix’ (or POSIX) time; origin is set to 1970-01-01.
+        If ‘julian’, unit must be ‘D’, and origin is set to beginning of Julian
+        Calendar. Julian day number 0 is assigned to the day starting at noon on
+        January 1, 4713 BC.
+        (default='unix')
```

Comment:
I'm ok with str support, but, it would be better if we also had support for 'gps'.

Something like
```
if epoch_origin == 'gps':
    epoch_origin = pds.datetime(1980, 1, 6)
```

Reply:
Right now, this is just the strings supported in `pandas.to_datetime`.  We can expand scope and add 'gps', or drop discussion of strings.  Not sure which way to go.  @aburrell?

Reply:
I like the idea of adding a GPS string, but think the best way to do that would be to create a time utility that has more epoch strings to supplement the pandas strings in `to_datetime`.  *Or*, even better, put in a pull request to Pandas with the GPS epoch.  I don't like the idea of adding the GPS string option to I/O input, since it's not an intuitive place to look for it.

Reply:
Could just be my personal style but I've tended towards dealing with time as soon as possible which has put it at i/o level.

If applied at a higher level the docstring for the time conversion utility would need to be explicit that it is actually changing the time, not just changing the epoch. Typically an epoch change would have a corresponding change in elapsed seconds so that the reported time remains the same.

Reply:
does gps time include leap seconds?  

Currently this is set up as a pass-through to the pandas routines, with docstring info partially copied from there (and updated to be less confusing).  We could just say that pysat always treats the origin as a datetime object, which is more restrictive than what pandas allows, but perhaps less confusing for end users on the pysat side.

Reply:
A GPS time routine seems... out of scope for this pull.  Changing the epoch is simple enough and I say we just use whatever pandas allows.

Reply:
> does gps time include leap seconds?
> 

No leap seconds for GPS.  All days are 86400 seconds.

> A GPS time routine seems... out of scope for this pull.

 I'm ok with it being handled later. 

> Adding @rstoneback into the discussion since he may know of other use cases I am missing.

My original gps comment wasn't attached to the right question. Sorry about that. CDAAC uses gps as their primary time across missions so it would be good if we had support.

Reply:
CDAAC was the inspiration for this pull, but putting in a custom datetime is all we need to get the basic data working there.  We should probably build some support mechanisms for this, but I think that is a separate issue.  The leap seconds concern me.

Reply:
```suggestion
```

Reply:
Leap seconds are a total pain.